src/librustc/ty/adjustment.rs

```diff
@@ -106,6 +107,10 @@ pub struct AutoDerefRef<'tcx> {
 impl<'tcx> AutoAdjustment<'tcx> {
     pub fn is_identity(&self) -> bool {
         match *self {
+            AdjustEmptyToAny(ref ty) => match ty.sty {
```

Comment:
You should never match against `AdjustEmptyToAny` with `ref`, `Ty` is a pointer itself.